Assets/Scripts/UI/Login/UILoginMenu.cs

```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
         /// <returns>True if input should be handled, false if not.</returns>
         private static bool ShouldInputBeHandled()
         {
-            // TODO: Clarify why this is only evaluated for PS4 and PS5
-#if UNITY_PS4 || UNITY_PS5
-            // TODO: Simplify this conditional - it's sort of confusing to read despite doing exactly what we want
-            if (null == EventSystem.current || EventSystem.current.sendNavigationEvents == !EOSManager.Instance.IsOverlayOpenWithExclusiveInput())
-            {
-                return true;
-            }
-
             bool shouldHandle = !EOSManager.Instance.IsOverlayOpenWithExclusiveInput();
-                
-            Debug.Log($"Input {(shouldHandle ? "enabled" : "disabled")} due to EOS Overlay.");
+
 #if ENABLE_INPUT_SYSTEM
             EventSystem.current.currentInputModule.enabled = shouldHandle;
+            EventSystem.current.sendNavigationEvents = shouldHandle;
 #else
+            // TODO: Clarify why this is only evaluated for PS4 and PS5
+#if UNITY_PS4 || UNITY_PS5
             EventSystem.current.sendNavigationEvents = shouldHandle;
 #endif
+#endif
 
-            return shouldHandle;
-#else
-            return true;
+#if ENABLE_DEBUG_INPUT
+            LogInputChanged(shouldHandle);
 #endif
+            return shouldHandle;
+
         }
 
+#if ENABLE_DEBUG_INPUT
+        static bool previousShouldHandle = false;
+        static void LogInputChanged(bool shouldHandle)
+        {
+            bool result = previousShouldHandle != shouldHandle ? true : false;
+            if (result)
+            {
+                Debug.LogWarning($"Input {(shouldHandle ? "enabled" : "disabled")} for main app");
```

Comment:
I've found that for clarity it can be useful to (instead of having an inline ternary statement) rephrase the log so it makes sense for the value `true` or `false` to be used. Something like:
```C#
Debug.LogWarning($"Should input be handled: {shouldHandle}.");
```

This is _not_ a critique relevant exactly for this PR - since you merely relocated the log line - and (if memory serves) I'm the one who originally wrote it 😆